swift/schema.py

```diff
@@ -1361,3 +1361,31 @@ class BorrowExpr(IdentityExpr):
     ```
     """
     pass
+
+@qltest.test_with('MacroDecl')
+class MacroRole(AstNode):
+    """
+    The role of a macro, for example #freestanding(declaration) or @attached(member).
+    """
+    kind: int | doc("kind of this macro role (declaration, expression, member, etc.)")
+    macro_syntax: int | doc("#freestanding or @attached")
```

Comment:
these should probably be marked `| ql.internal`: we generally don't want to expose magic enumeration numbers, and we usually hide them either behind predicates (for example `isDeclarationMacro` etc and `isAttached/isFreestanding`)

Reply:
I don't mind either way. Should I hide these and add predicates for all the cases?

Reply:
I'd say yes, @MathiasVP maybe you have something to say?

Reply:
I agree with @redsun82 👍

Reply:
@redsun82 adding `ql.internal` doesn't seem to have any effect on the generated files, and I don't see it used anywhere except for top-level decls in the schema file.

Reply:
ah, indeed, I thought I had that already implemented, but I was wrong

Reply:
I've implemented it in https://github.com/github/codeql/pull/14902 but feel free to ignore it for this PR :slightly_smiling_face: